server/hedley/modules/custom/hedley_admin/scripts/generate-demographics-report.php

```diff
@@ -0,0 +1,389 @@
+}
+$group_encounter_all = group_encounter_all();
+$group_encounter_unique = group_encounter_unique();
+drush_print("Ready to print the report.\n\n");
+
+drush_print("# Demographics report  - " . date('D/m/Y'));
+
+drush_print("## REGISTERED PATIENTS");
+
+$registered = [
+  [
+    '0 - 1M',
+    classified_count('lt1m', 'male'),
+    classified_count('lt1m', 'female'),
+  ],
+  [
+    '1M - 2Y',
+    classified_count('lt2y', 'male'),
+    classified_count('lt2y', 'female'),
+  ],
+  [
+    '2Y - 5Y',
+    classified_count('lt5y', 'male'),
+    classified_count('lt5y', 'female'),
+  ],
+  [
+    '5Y - 10Y',
+    classified_count('lt10y', 'male'),
+    classified_count('lt10y', 'female'),
+  ],
+  [
+    '10Y - 20Y',
+    classified_count('lt20y', 'male'),
+    classified_count('lt20y', 'female'),
+  ],
+  [
+    '20Y - 50Y',
+    classified_count('lt50y', 'male'),
+    classified_count('lt50y', 'female'),
+  ],
+  [
+    '50Y +',
+    classified_count('mt50y', 'male'),
+    classified_count('mt50y', 'female'),
+  ],
+];
+$text_table = new HedleyAdminTextTable(['Registered', 'Male', 'Female']);
+$text_table->addData($registered);
+
+drush_print($text_table->render());
+
+$impacted = [
+  [
+    '0 - 1M',
+    impacted_count('lt1m', 'male'),
+    impacted_count('lt1m', 'female'),
+  ],
+  [
+    '1M - 2Y',
+    impacted_count('lt2y', 'male'),
+    impacted_count('lt2y', 'female'),
+  ],
+  [
+    '2Y - 5Y',
+    impacted_count('lt5y', 'male'),
+    impacted_count('lt5y', 'female'),
+  ],
+  [
+    '5Y - 10Y',
+    impacted_count('lt10y', 'male'),
+    impacted_count('lt10y', 'female'),
+  ],
+  [
+    '10Y - 20Y',
+    impacted_count('lt20y', 'male'),
+    impacted_count('lt20y', 'female'),
+  ],
+  [
+    '20Y - 50Y',
+    impacted_count('lt50y', 'male'),
+    impacted_count('lt50y', 'female'),
+  ],
+  [
+    '50Y +',
+    impacted_count('mt50y', 'male'),
+    impacted_count('mt50y', 'female'),
+  ],
+];
+$text_table = new HedleyAdminTextTable(['Impacted (2+ visits)', 'Male', 'Female']);
+$text_table->addData($impacted);
+
+drush_print($text_table->render());
+
+drush_print("## ENCOUNTERS");
+
+/**
+ * Gathers group encounter visits by type.
+ *
+ * @return array
+ */
+function group_encounter_all() {
+  return db_query("
+  SELECT
+  field_group_type_value as type, COUNT(*) as counter
+FROM
+  (
+    SELECT
+      field_group_type_value,
+      p.field_person_target_id,
+      sess_rel.field_session_target_id
+    FROM
+      field_data_field_session sess_rel
+        LEFT JOIN field_data_field_clinic c ON sess_rel.field_session_target_id = c.entity_id
+        LEFT JOIN field_data_field_group_type gt ON field_clinic_target_id = gt.entity_id
+        LEFT JOIN field_data_field_person p ON p.entity_id = sess_rel.entity_id
+        LEFT JOIN person_classified class ON p.field_person_target_id = class.entity_id
+    WHERE
+        sess_rel.bundle IN ('attendance', 'birth_plan', 'breast_exam', 'child_fbf', 'contributing_factors',
```

Comment:
@AronNovak 

Wondering what this list is meant to represent, and, if there's no way to generate it dynamically, as new content types are frequently added in the process of develoment.

Reply:
As it was a pure SQL script, that's why it was a must to hard-code, but as now it's a hybrid SQL + PHP, I refactor these to be dynamic, it would not slow down anything.